src/Parallel/Algorithm.hpp

```diff
@@ -720,112 +879,96 @@ AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>::
-              using first_databox = tmpl::at_c<databox_types_this_phase, 0>;
-              using last_databox =
-                  tmpl::at_c<databox_types_this_phase,
-                             tmpl::size<databox_types_this_phase>::value - 1>;
-              using local_this_action =
-                  tmpl::at_c<actions_list, decltype(current_iter)::value>;
-              if (box_.which() ==
-                  static_cast<int>(
-                      tmpl::index_of<variant_boxes, first_databox>::value)) {
-                using this_databox = first_databox;
-                auto& box = boost::get<this_databox>(box_);
-                if (not check_if_ready(
-                        Algorithm_detail::is_is_ready_callable_t<
-                            local_this_action, this_databox,
-                            tuples::tagged_tuple_from_typelist<inbox_tags_list>,
-                            Parallel::ConstGlobalCache<metavariables>,
-                            array_index>{},
-                        local_this_action{}, box)) {
-                  take_next_action = false;
-                  return nullptr;
-                }
-                performing_action_ = true;
-                algorithm_step_++;
-                invoke_this_action(
-                    box,
-                    typename std::tuple_size<decltype(local_this_action::apply(
-                        box, inboxes_, *const_global_cache_,
-                        std::as_const(array_index_), actions_list{},
-                        std::add_pointer_t<ParallelComponent>{}))>::type{});
-              } else if (box_.which() ==
-                         static_cast<int>(
-                             tmpl::index_of<variant_boxes,
-                                            last_databox>::value)) {
-                using this_databox = last_databox;
-                auto& box = boost::get<this_databox>(box_);
-                if (not check_if_ready(
-                        Algorithm_detail::is_is_ready_callable_t<
-                            local_this_action, this_databox,
-                            tuples::tagged_tuple_from_typelist<inbox_tags_list>,
-                            Parallel::ConstGlobalCache<metavariables>,
-                            array_index>{},
-                        local_this_action{}, box)) {
-                  take_next_action = false;
-                  return nullptr;
-                }
-                performing_action_ = true;
-                algorithm_step_++;
-                invoke_this_action(
-                    box,
-                    typename std::tuple_size<decltype(local_this_action::apply(
-                        box, inboxes_, *const_global_cache_,
-                        std::as_const(array_index_), actions_list{},
-                        std::add_pointer_t<ParallelComponent>{}))>::type{});
-              } else {
-                display_databox_error(__LINE__);
-              }
+          // clang-format on
```

Comment:
Uh oh, do I correctly see that most of this diff is formatting? Can I be a pain and request you move the whitespace changes to a separate commit? (and below)

Reply:
I've done one better and entirely removed all of the make_overloader business from this file, converting to far more succinct `if constexpr`s.